### ISSUE TITLE
[BUG] initialTopMostItemIndex race condition in Safari

### ISSUE THREAD

**Issue by ignatevdev** [petyosi/react-virtuoso#874]
**Describe the bug**
When using `initialTopMostItemIndex` along with `alignToBottom` the initial scroll doesn't work reliably in Safari due to some race condition between items render and initial scroll.  This happens only in Safari, Chrome / Firefox don't have that issue.

After debugging I found that scroll element's `scrollHeight` is not correct at the moment when initial scroll is firing.
https://github.com/petyosi/react-virtuoso/blob/896bdaf40ce5939eab3b3adca37fa023875d37fc/src/hooks/useScrollTop.ts#L93

Following this code, it seems that virtuoso relies on the list items to be rendered immediately after mount and uses `setTimeout` without timing to postpone the scroll. However, in some cases in Safari that's not enough and the initial scroll breaks. 
https://github.com/petyosi/react-virtuoso/blob/master/src/initialTopMostItemIndexSystem.ts#L41

Adding a timing to this `setTimeout` bigger than 20 solves the issue, but I'm not sure if that's the right way to fix it.

**Reproduction**
I wasn't able to reproduce this in a sandbox, probably because it only happens on multiple re-renders when `<Virtuoso />` is rendered conditionally only after the data has loaded, but I'll try again if necessary. 

**Expected behavior**
Scroll to `initialTopMostItemIndex` working correctly in Safari.

**Screenshots**
https://user-images.githubusercontent.com/8458382/229528719-2831c66d-c8aa-4d52-9a6b-f89b1059217d.mov

**Desktop:**
 - OS: OS X
 - Browser: Safari 15.6.1
 - React: 18.2.0
 - Virtuoso: 4.1.1

**Comment by petyosi** [comment]
I do need a reproduction of the problem to make this actionable - from the video it looks like it's a fairly vanilla case? I'm ok if the example needs a few refreshes to break - still something I can work with. 

Also, ensure that you test in the latest Safari (16.4), as it has quite a few improvements and bugfixes that are known to affect the component. 

I am closing this as, more often than not, people don't come back to me with anything helpful. But I will be happy to reopen once there's a meaningful follow up.
